pkg/labels/integration_test.go

```diff
@@ -332,3 +338,66 @@ func TestRemoveLabelsTxnFailsIfLabelsChange(t *testing.T) {
 		t.Fatal("expected the transaction to fail when the labels are changed out from under it")
 	}
 }
+
+func TestMutateAndSelectHTTPApplicator(t *testing.T) {
```

Comment:
sometimes I wish there were a way to see what changed when the lines were moved from file to file.

I had to diff it manually to answer the question

```diff
        }
 }

-func TestMutateAndSelect(t *testing.T) {
-       server, applicator := fakeServerAndApplicator(t, 0)
+func TestMutateAndSelectHTTPApplicator(t *testing.T) {
+       fixture := consulutil.NewFixture(t)
+       labelServer := NewHTTPLabelServer(NewConsulApplicator(fixture.Client, 0), 0, logging.TestLogger())
+       server := httptest.NewServer(labelServer.Handler())
        defer server.Close()
+       url, err := url.Parse(server.URL)
+       Assert(t).IsNil(err, "expected no error parsing url")
+
+       applicator, err := NewHTTPApplicator(nil, url)
+       Assert(t).IsNil(err, "expected no error creating HTTP applicator")
+
        podID := "abc/123"
        colorLabel := "p2/color"

@@ -159,14 +167,14 @@ func TestMutateAndSelect(t *testing.T) {
        podLabels, err := applicator.GetLabels(POD, podID)
        Assert(t).IsNil(err, "Should not have erred getting labels for the pod")
        Assert(t).AreEqual("red", podLabels.Labels.Get(colorLabel), "Should have seen red on the color label")
-       matches, err := applicator.GetMatches(labels.Everything().Add(colorLabel, labels.EqualsOperator, []string{"red"}), POD, false)
+       matches, err := applicator.GetMatches(klabels.Everything().Add(colorLabel, klabels.EqualsOperator, []string{"red"}), POD, false)
        Assert(t).IsNil(err, "There should not have been an error running a selector")
        Assert(t).AreEqual(1, len(matches), "Should have gotten a match")
        Assert(t).AreEqual(podID, matches[0].ID, "Wrong pod returned")
        Assert(t).AreEqual("red", matches[0].Labels.Get(colorLabel), "Wrong color returned")

        // set all labels, expect all to change
-       Assert(t).IsNil(applicator.SetLabels(POD, podID, labels.Set{colorLabel: "green", "state": "experimental"}), "Should not err setting labels")
+       Assert(t).IsNil(applicator.SetLabels(POD, podID, klabels.Set{colorLabel: "green", "state": "experimental"}), "Should not err setting labels")
        podLabels, err = applicator.GetLabels(POD, podID)
        Assert(t).IsNil(err, "Should not have erred getting labels for the pod")
        Assert(t).AreEqual("green", podLabels.Labels.Get(colorLabel), "Should have seen green on the color label")
```